uhabits-core/src/jvmTest/java/org/isoron/uhabits/core/BaseUnitTest.kt

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (C) 2016-2021 Álinson Santos Xavier <git@axavier.org>
+ *
+ * This file is part of Loop Habit Tracker.
+ *
+ * Loop Habit Tracker is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * Loop Habit Tracker is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.isoron.uhabits.core
+
+import org.apache.commons.io.IOUtils
+import org.isoron.uhabits.core.commands.CommandRunner
+import org.isoron.uhabits.core.database.Database
+import org.isoron.uhabits.core.database.DatabaseOpener
+import org.isoron.uhabits.core.database.JdbcDatabase
+import org.isoron.uhabits.core.database.MigrationHelper
+import org.isoron.uhabits.core.models.HabitList
+import org.isoron.uhabits.core.models.ModelFactory
+import org.isoron.uhabits.core.models.Timestamp
+import org.isoron.uhabits.core.models.memory.MemoryModelFactory
+import org.isoron.uhabits.core.tasks.SingleThreadTaskRunner
+import org.isoron.uhabits.core.test.HabitFixtures
+import org.isoron.uhabits.core.utils.DateUtils.Companion.getStartOfTodayCalendar
+import org.isoron.uhabits.core.utils.DateUtils.Companion.setFixedLocalTime
+import org.isoron.uhabits.core.utils.DateUtils.Companion.setStartDayOffset
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito
+import org.mockito.junit.MockitoJUnitRunner
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.io.IOException
+import java.io.InputStream
+import java.nio.file.Paths
+import java.sql.DriverManager
+import java.sql.SQLException
+
+@RunWith(MockitoJUnitRunner::class)
+open class BaseUnitTest {
+    protected open var habitList: HabitList? = null
+    protected var fixtures: HabitFixtures? = null
+    protected var modelFactory: ModelFactory? = null
+    protected var taskRunner: SingleThreadTaskRunner? = null
+    protected open var commandRunner: CommandRunner? = null
```

Comment:
I think the majority of these are initialized in `setUp`, so they could be `lateinit var` instead